Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py

```diff
@@ -649,49 +674,62 @@ def fetch_incidents(client: Client, max_results: int, last_run: Dict[str, int],
-            incident_created_time_ms = incident_created_time * 1000
-
-            # to prevent duplicates, we are only adding incidents with creation_time >= last fetched incident
-            if last_fetch:
-                if incident_created_time < last_fetch:
-                    continue
-
-            incidents.append({
-                'name': a.get('name', 'No name') + ' - ' + insight_id,
-                'occurred': timestamp_to_datestring(incident_created_time_ms),
-                'details': a.get('description'),
-                'severity': translate_severity(a.get('severity')),
-                'rawJSON': json.dumps(a)
-            })
-            current_fetch_ids.append(insight_id)
-
-            # Update last run and add incident if the incident is newer than last fetch
-            if incident_created_time > latest_created_time:
-                latest_created_time = incident_created_time
+    hasNextPage = True
+    while hasNextPage:
+
+        # only query parameter that changes loop to loop is the offset
+        query['offset'] = str(offset)
+        resp_json = client.req('GET', 'sec/v1/insights', query)
+        for a in resp_json.get('objects'):
+
+            # If no created_time set is as epoch (0). We use time in ms so we must
+            # convert it from the API response
+            insight_timestamp = a.get('created')
+            insight_id = a.get('id')
+            if insight_id and insight_timestamp and insight_id not in last_fetch_ids:
+                try:
+                    incident_datetime = datetime.strptime(insight_timestamp, '%Y-%m-%dT%H:%M:%S.%f')
+                except ValueError:
+                    incident_datetime = datetime.strptime(insight_timestamp, '%Y-%m-%dT%H:%M:%S')
+
+                incident_created_time = int((incident_datetime - datetime.utcfromtimestamp(0)).total_seconds())
+                incident_created_time_ms = incident_created_time * 1000
+
+                # to prevent duplicates, we are only adding incidents with creation_time >= last fetched incident
+                if last_fetch:
+                    if incident_created_time < last_fetch:
+                        continue
+
+                incidents.append({
+                    'name': a.get('name', 'No name') + ' - ' + insight_id,
+                    'occurred': timestamp_to_datestring(incident_created_time_ms),
+                    'details': a.get('description'),
+                    'severity': translate_severity(a.get('severity')),
+                    'rawJSON': json.dumps(a)
+                })
+                current_fetch_ids.append(insight_id)
+
+                # Update last run and add incident if the incident is newer than last fetch
+                if incident_created_time > latest_created_time:
+                    latest_created_time = incident_created_time
+
+        if not resp_json.get('hasNextPage') or len(incidents) >= resp_json.get('total'):
```

Comment:
```python
len(incidents) >= resp_json.get('total'):
```
might raise `TypeError: '>=' not supported between instances of 'int' and 'NoneType'`
i recommend being a bit more protective here

Reply:
Done